drivers/include/pn532.h

```diff
@@ -0,0 +1,373 @@
+typedef enum {
+    ISO14443A_UNKNOWN,
+    ISO14443A_MIFARE,
+    ISO14443A_TYPE4
+} nfc_iso14443a_type_t;
+
+/**
+ * @brief ISO14443A tag description
+ */
+typedef struct {
+    char sns_res;
+    char sel_res;
+    char target;
+    char auth;
+    char id_len;
+    nfc_iso14443a_type_t type;
+    char id[8];
+} nfc_iso14443a_t;
+
+/**
+ * @brief Mifare keys
+ */
+typedef enum {
+    PN532_MIFARE_KEY_A = 0x60,
+    PN532_MIFARE_KEY_B = 0x61
+} pn532_mifare_key_t;
+
+/**
+ * @brief Obtain Tag 4 data length from buffer
+ *
+ * This is useful in case the length has been read and one intents to read the
+ * data.
+ */
+#define PN532_ISO14443A_4_LEN_FROM_BUFFER(b) ((b[0] << 8) | b[1])
+
+/**
+ * @brief Hard reset the chipset
+ *
+ * The chipset is reset by toggling the reset pins
+ *
+ * @param[in]  dev          target device
+ *
+ */
+void pn532_reset(pn532_t *dev);
+
+/**
+ * @brief Initialize the module and peripherals
+ *
+ * This is the first method to be called in order to interact with the pn532.
+ * It configures the GPIOs and the i2c interface.
+ *
+ *  @param[in]  dev         target device
+ *  @param[in]  params      configuration parameters
+ *
+ * @return                  0 on success
+ */
+int pn532_init(pn532_t *dev, const pn532_params_t *params);
+
+/**
+ * @brief Get the firmware version of the pn532
+ *
+ * The firmware version returned is a 4 byte long value:
+ *  - ic version,
+ *  - fw version,
+ *  - fw revision
+ *  - feature support
+ *
+ * @param[in]  dev          target device
+ * @param[out] fw_ver       encoded firmware version
+ *
+ * @return                  0 on success
+ */
+int pn532_fw_version(pn532_t *dev, uint32_t *fw_ver);
+
+/**
+ * @brief Read register of the pn532
+ *
+ * Refer to the datasheet for a comprehensive list of registers and meanings.
+ * For SFR registers the high byte must be set to 0xff.
+ *
+ * @param[in]  dev          target device
+ * @param[out] out          value of the register
+ * @param[in]  addr         address of the register to read
+ *
+ * @return                  0 on success
+ */
+int pn532_read_reg(pn532_t *dev, char *out, unsigned addr);
+
+/**
+ * @brief Write register of the pn532
+ *
+ * Refer to the datasheet for a comprehensive list of registers and meanings.
+ *
+ * @param[in]  dev          target device
+ * @param[in]  addr         address of the register to read
+ * @param[in]  val          value to write in the register
+ *
+ * @return                  0 on success
+ */
+int pn532_write_reg(pn532_t *dev, unsigned addr, char val);
+
+/**
+ * @brief Read status of the GPIOs
+ *
+ * 3 byte encoded values of p3, p7 and l0l1. Bitwise information as follows:
+ *  - p3:   p30, p31, p32, p33, p34, p35, 0, 0
+ *  - p7:     0, p71, p72,   0,   0,   0, 0, 0
+ *  - l0l1:  l0, l1
+ *
+ * @param[in]  dev          target device
+ * @param[out] gpio         status of the gpio ports
+ *
+ * @return                  0 on success
+ */
+int pn532_read_gpio(pn532_t *dev, uint32_t *gpio);
```

Comment:
This function is however in the test/example never used, it allows to control some GPIOs exposed in the PN532. I've no preference on that, I can remove it :+1: